### PR TITLE
[DOC] Remove GLM experimental warnings and references to nistats 

### LINKED ISSUE #3447: Remove notes in docstrings of `nilearn.glm` classes and functions saying they are experimental
Now that we removed the FutureWarning saying that `nilearn.glm` is experimental I think we can also remove the notes in the docstrings of the classes/functions under this module that say that they are experimental

### Affected pages
e.g. 
https://nilearn.github.io/stable/modules/generated/nilearn.glm.Contrast.html#nilearn.glm.Contrast
https://nilearn.github.io/stable/modules/generated/nilearn.glm.compute_contrast.html#nilearn.glm.compute_contrast

### PR DESCRIPTION
**Pull request by ymzayek**
Closes #3447 .

Changes proposed in this pull request:

- Remove notes in docstrings warning the glm module is experimental
- Remove references to nistats in glm

### CODE REVIEW

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @ymzayek !

**Review by bthirion — COMMENTED** [review]
LGTM so far. 
Maybe also remove the "nistats to nilearn" transition paragraph in Nilearn's documentation.

**Comment by ymzayek** [comment]
@bthirion I removed reference to nistats migration guide on the landing page as well as the link under the 'Development' menu on the left hand side. The file `nistats_migration.rst` is still in the code base but is not visible to the user and can only be accessed by specifying the file name in the URL. If you think it should be visible then I can add it back to the side menu otherwise would it hurt to get rid of it altogether?

**Comment by bthirion** [comment]
I propose to completely get rid of it. Nobody remembers nistats nowadays.

File: doc/changes/0.7.0.rst [hunk 1]
```diff
@@ -34,7 +34,7 @@ HIGHLIGHTS
 NEW
 ---
 - Nilearn now includes the functionality of `Nistats <https://nistats.github.io>`_.
-  :ref:`Here's a guide to replacing Nistats imports to work in Nilearn. <nistats_migration>`
+  `Here's a guide to replacing Nistats imports to work in Nilearn. <nistats_migration>`
```
Comment (bthirion) [inline doc/changes/0.7.0.rst:37]: This should be removed, right ?

**Review by bthirion — COMMENTED** [review]
LGTM otherwise, thx !

**Review by bthirion — APPROVED** [review]
LGTM, thx!

**Comment by ymzayek** [comment]
Merging. Failures are unrelated

**Comment by bthirion** [comment]
Great, thx !

**Pull request merged**